tests/test_sensitivity_matrix.py

```diff
@@ -0,0 +1,148 @@
+# This file is part of ts_ofc.
+#
+# Developed for Vera Rubin Observatory.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import unittest
+from glob import glob
+from pathlib import Path
+
+import numpy as np
+import yaml
+from lsst.ts.ofc import OFCData, SensitivityMatrix
+
+
+class TestSensitivityMatrix(unittest.TestCase):
+    """Test the SensitivityMatrix class."""
+
+    def setUp(self):
+        self.ofc_data = OFCData("lsst")
+
+        self.sensitivity_matrix = SensitivityMatrix(self.ofc_data)
+
+        self.field_angles = [self.ofc_data.gq_points[sensor] for sensor in range(31)]
+        self.unrotated_sensitivity_matrix = self.sensitivity_matrix.evaluate(
+            field_angles=self.field_angles
+        )
+
+        gq_sensitivity_matrix_file = Path(
+            glob(
+                str(
+                    Path(__file__).parent.absolute()
+                    / "testData"
+                    / "legacy"
+                    / f"lsst_{self.ofc_data.sen_m_filename_root}*.yaml"
+                )
+            )[0]
+        )
+
+        with open(gq_sensitivity_matrix_file, "r") as fp:
+            self.gq_sensitivity_matrix = np.array(yaml.safe_load(fp))
+
+            # Note that we use only up to 31 index for the old
+            # sensitivity matrix, because the last 4 elements
+            # corresponded to the corner wavvefront sensors
+            # which we have excluded from the unrotated_sensitivity_matrix.
+            self.gq_sensitivity_matrix = self.gq_sensitivity_matrix[:31, ...]
+
+    def mean_squared_residual(self, new_array, reference_array):
+        return np.sum((new_array - reference_array) ** 2) / np.sum(reference_array**2)
+
+    def test_dz_sensitivity_matrix(self):
+        print(self.ofc_data.gq_points)
+        # Check that the sensitivity matrix for dz is the same as the one
+        # that we had before the last ts_ofc refactor.
+        for dof in np.arange(50):
+            print(dof)
+            print(
+                self.mean_squared_residual(
+                    self.unrotated_sensitivity_matrix[..., dof],
+                    self.gq_sensitivity_matrix[..., dof],
+                )
+            )
```

Comment:
Please remove these `print()` functions. If you want to check something, use the assertion method in the unit tests to make sure the system will complain if the values are not expected.

Reply:
Thanks for the catch!